File: App003_UnitConverter/app/src/main/java/com/example/app03_unitconverter/MainActivity.java
package com.example.app03_unitconverter;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import java.text.DecimalFormat;

public class MainActivity extends AppCompatActivity {

    EditText inputField;
    Button submitButton;
    TextView textView;
    double factor = 2.20462;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        submitButton = findViewById(R.id.submit_button);
        inputField = findViewById(R.id.text_input);
        textView = findViewById(R.id.textView);
        textView.setText("0");


        submitButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String input = inputField.getText().toString();
                double kilos = Double.parseDouble(input);
                double pounds = kilos * factor;
                DecimalFormat decimalFormat = new DecimalFormat("#.00000");
                String stringPounds = decimalFormat.format(pounds);
                textView.setText(String.valueOf(pounds));
            }
        });
    }
}

File: App026_Kt_UnitConverter/app/src/main/java/com/kes/app026_kt_unitconverter/MainActivity.kt
package com.kes.app026_kt_unitconverter

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        val inputView: EditText = findViewById(R.id.input)
        val outputView: TextView = findViewById(R.id.result)
        val button: Button = findViewById(R.id.submitBtn)

        button.setOnClickListener() {
            val kilos: Double = inputView.text.toString().toDouble();
            val pounds: Double = convertToPounds(kilos);
            outputView.text = pounds.toString();
        }

    }

    fun convertToPounds(kilos: Double): Double = kilos * 2.20462

}